spec/models/slot_availability_validation_spec.rb

```diff
@@ -0,0 +1,186 @@
+require 'rails_helper'
+
+RSpec.describe SlotAvailabilityValidation, type: :model do
+  subject { described_class.new(visit: visit) }
+
+  let(:prison) { build_stubbed(:prison) }
+  let(:visit) { build_stubbed(:visit, prison: prison) }
+  let(:slot1) do
+    date = 6.days.from_now.to_date
+    ConcreteSlot.new(date.year, date.month, date.day, 14, 30, 15, 30)
+  end
+  let(:slot2) do
+    date = 5.days.from_now.to_date
+    ConcreteSlot.new(date.year, date.month, date.day, 14, 30, 15, 30)
+  end
+  let(:slot3) do
+    date = 4.days.from_now.to_date
+    ConcreteSlot.new(date.year, date.month, date.day, 14, 30, 15, 30)
+  end
+  let(:requested_slots) { [slot1, slot2, slot3] }
+  let(:api_slots) do
+    available_slots.map do |slot|
+      { 'time' => slot.to_s, 'max_adults' => anything }
+    end
+  end
+
+  before do
+    allow(visit).to receive(:slots).and_return(requested_slots)
+  end
+
+  describe 'when the NOMIS API is disabled' do
+    before do
+      allow(Nomis::Api).to receive(:enabled?).and_return(false)
+      subject.valid?
+    end
+
+    it "doesn't add an error to the slots" do
+      requested_slots.each do |slot|
+        expect(subject.errors[slot.to_s]).to be_empty
+      end
+    end
+
+    it 'is unknown result' do
+      expect(subject).to be_unknown_result
+    end
+  end
+
+  describe 'when the NOMIS API is enabled' do
+    context 'and the api returns an error' do
+      before do
+        expect_any_instance_of(Nomis::Client).
+          to receive(:get).and_raise(Nomis::APIError)
+      end
+
+      it 'adds no errors for any slot' do
+        is_expected.to be_valid
+
+        requested_slots.each do |slot|
+          expect(subject.errors[slot.to_s]).to be_empty
+        end
+      end
+
+      it 'is unknown result' do
+        subject.valid?
+
+        expect(subject).to be_unknown_result
+      end
+    end
+
+    context 'and working correctly with valid slots' do
+      before do
+        allow(Nomis::Api).to receive(:enabled?).and_return(true)
+
+        expect_any_instance_of(Nomis::Api).
```

Comment:
The connection pool of nomis api clients means we don't know exactly what instance it will be, if I remember correctly